### PR TITLE
Fixes invalid assert

### PR DESCRIPTION
**Pull request by boudewijn-tribler**
See #155.

### CODE REVIEW

File: database.py [hunk 1]
```diff
@@ -361,7 +361,8 @@ def executemany(self, statement, sequenceofbindings):
             assert isinstance(statement, unicode), "The SQL statement must be given in unicode"
             assert isinstance(sequenceofbindings, (tuple, list, set)), "The sequenceofbindings must be a tuple, list, or set"
             assert all(isinstance(x, (tuple, list, dict, set)) for x in list(sequenceofbindings)), "The sequenceofbindings must be a list with tuples, lists, dictionaries, or sets"
-            assert not filter(lambda x: filter(lambda y: isinstance(y, str), x), list(sequenceofbindings)), "The bindings may not contain a string. \nProvide unicode for TEXT and buffer(...) for BLOB."
+            assert all(all(not isinstance(y, str) for y in x) for x in sequenceofbindings), "The bindings may not contain a string. \nProvide unicode for TEXT and buffer(...) for BLOB."
```
Comment (NielsZeilemaker) [inline database.py]: This assert now causes a dict to have non-string keys for namedparameter queries.

File: database.py [hunk 2]
```diff
@@ -340,9 +347,9 @@ def executemany(self, statement, sequenceofbindings):
         @param statement: the SQL statement that is to be executed.
         @type statement: unicode
 
-        @param bindings: a sequence of values that must be set to the placeholders in statement.
-         Each element in sequence is another tuple containing bindings.
-        @type bindings: list containing tuples
+        @param sequenceofbindings: a list, tuple or set of bindings, where every binding contains
+        the values that must be set to the placeholders in statement.
+        @type bindings: list, tuple or set
```
Comment (NielsZeilemaker) [inline database.py]: or dict

Comment (boudewijn-tribler) [inline database.py]: the sequenceofbindings can not be a dict, although the sequence can contain dictionaries.

the `@type bindings` should be `@type sequenceofbindings`, as there is no parameter called bindings.  And sequenceofbindings can also be a generator.

**Comment by boudewijn-tribler** [comment]
segfaults.  Elric did a system upgrade, lets see if it helps.

**Comment by boudewijn-tribler** [comment]
retest this please

**Pull request merged**